### ISSUE TITLE
Building 32bit RPM on x86_64 system

### ISSUE THREAD

**Issue by benjoellush** [jordansissel/fpm#835]
The comment for the "-a" arg in FPM:

-a, --architecture ARCHITECTURE The architecture name. Usually matches 'uname -m'. For automatic values, you can use '-a all' or '-a native'. These two strings will be translated into the correct value for your platform and target package type.

I need to do -i386 and have the dependencies also be i386

Is there a way to:

fpm -s none -t rpm -n my_meta_package_32bit -v 1.2.3 -a i386 -d "sqlite(i386) = 3.3.6"
..
Created rpm {:path=>"my_meta_package_32bit-1.2.3-1.i386.rpm"}

**Comment by dhajoshi** [comment]
+1

**Comment by phathocker** [comment]
+1

**Comment by jordansissel** [comment]
Perhaps a silly question, but does `-a i386` not work to build a 32bit rpm?

**Comment by dhajoshi** [comment]
No, -a works but the question is about setting up dependencies and that dependent RPM should be of 32bit or 64 ( basically with -d ) , i think i got some solution before but i have to re-collect.

**Comment by miked63017** [comment]
I may be wrong, but I thought the dependencies are actually resolved at
install time, meaning that the system you install the rpm on should know if
its 32 or 64 bit and install the packages accordingly. We have many CentOS
systems at work and to install a 32 bit rpm on a 64 bit system takes a
little extra work, so in other words it automatically does whats right
because yum is smart enough to know its own architecture. I can't speak at
all for debian or other systems besides yum on that though.

**Comment by phathocker** [comment]
My issue seems to be resolved.  I was getting an error when installing the RPM using ```rpm -ivh``` or ```yum localinstall``` but when I uploaded to my yum repo, the issue went away.

**Comment by dirkbaechle** [comment]
@benjoellush Sorry for hijacking this thread, but if you're still interested in helping out with SCons development (see OpenHatch, you didn't leave a public mail), join our user mailing list ( scons-users@scons.org , see also http://scons.org/lists.php ) and ask for directions there.
